### ISSUE TITLE
`if-then-else`/`needless_bool` is breaking easily

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/rust-clippy#11248]
### Summary

The `if-then-else` lint is breaking very easily.

### Reproducer

```rust
if x {
    // some comment
    true
} else {
    true
}
```

The `if-then-else` lint is not triggered if the block contains comments.

### Version

```text
I tested it in clippy repository directly.
```

**Comment by y21** [comment]
I'm assuming you mean the `needless_bool` lint?
The reproducer confusingly triggers the `if_same_then_else` lint, which tripped me up when I was trying to test this locally and made me think there's no issue here.
Maybe you should change one of the bool literals in the example to `false`, so that they aren't the same code and doesn't trigger a different lint

**Comment by GuillaumeGomez** [comment]
Indeed, I got the lint name wrong (updating the issue). Some more details: I have the issue because I'm migrating clippy ui tests to use the `//~` everywhere. And in this case, it simply breaks `needless_bool/simple.rs`.

**Comment by Alexendoo** [comment]
This is intentional #10766
